libevm/VMFactory.cpp

```diff
@@ -85,8 +85,15 @@ void setVMKind(const std::string& _name)
     }
 
     // If no match for predefined VM names, try loading it as an EVMC VM DLL.
+    g_kind = VMKind::DLL;
+
+    // Release previous instance
+    g_evmcDll.reset();
```

Comment:
With these two above, even if testeth would ignore the exceptions below, the factory will be in an expected failing state and the assertion in `VMFactory::create` will take of that.